addons/point_of_sale/static/src/js/Chrome.js

```diff
@@ -1,120 +1,141 @@
+        this.env = env;
+        this.__owl__.childEnv = env;
+        useSubEnv({
+            get isMobile() {
+                return window.innerWidth <= 768;
+            },
+        });
+        let currentIsMobile = this.env.isMobile;
+        const updateUI = debounce(() => {
+            if (this.env.isMobile !== currentIsMobile) {
+                currentIsMobile = this.env.isMobile;
+                this.render(true);
+            }
+        }, 15); // FIXME POSREF use throttleForAnimation?
+        useExternalListener(window, "resize", updateUI);
+        onWillUnmount(updateUI.cancel);
+        // END ChromeAdapter
+
         super.setup();
         useExternalListener(window, "beforeunload", this._onBeforeUnload);
         useListener("show-main-screen", this.__showScreen);
-        useListener("toggle-debug-widget", debounce(this._toggleDebugWidget, 100));
-        useListener("toggle-mobile-searchbar", this._toggleMobileSearchBar);
         useListener("show-temp-screen", this.__showTempScreen);
         useListener("close-temp-screen", this.__closeTempScreen);
         useListener("close-pos", this._closePos);
         useListener("loading-skip-callback", () => this.env.proxy.stop_searching());
-        useListener("play-sound", this._onPlaySound);
+        const sound = useService("sound");
+        useListener("play-sound", ({ detail: name }) => sound.play(name));
         useListener("set-sync-status", this._onSetSyncStatus);
         useListener("show-notification", this._onShowNotification);
         useListener("close-notification", this._onCloseNotification);
         useListener("connect-to-proxy", this.connect_to_proxy);
         useBus(this.env.posbus, "start-cash-control", this.openCashControl);
         NumberBuffer.activate();
 
-        this.state = useState({
-            uiState: "LOADING", // 'LOADING' | 'READY' | 'CLOSING'
-            debugWidgetIsShown: true,
-            mobileSearchBarIsShown: false,
-            hasBigScrollBars: false,
-            sound: { src: null },
-            notification: {
-                isShown: false,
-                message: "",
-                duration: 2000,
-            },
-            loadingSkipButtonIsShown: false,
-        });
+        this.state = usePos();
 
-        this.mainScreen = useState({ name: null, component: null });
+        this.mainScreen = this.state.mainScreen;
         this.mainScreenProps = {};
 
         this.tempScreen = useState({ isShown: false, name: null, component: null });
         this.tempScreenProps = {};
 
-        this.progressbar = useRef("progressbar");
-
-        this.previous_touch_y_coordinate = -1;
-
-        const pos = reactive(
-            this.env.pos,
-            batched(() => this.render(true))
-        );
-        useSubEnv({ pos });
+        useSubEnv({
+            pos: reactive(
+                this.env.pos,
+                batched(() => this.render(true)) // FIXME POSREF remove render(true)
```

Comment:
I'm leaving a bunch of `POSREF` comments for things that need to be addressed before we call the PoS refactoring complete but most of them probably won't be tackled in this specific PR. Hopefully once we've moved all of the PosGlobalState to the pos store we'll be able to just remove this and it will work as intended

Reply:
Don't worry, I'm not asking for change, just suggesting idea as well